src/main/java/com/mcmoddev/communitymod/erdbeerbaerlp/iconmod/ModClassWriter.java

```diff
@@ -0,0 +1,41 @@
+package com.mcmoddev.communitymod.erdbeerbaerlp.iconmod;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+
+import net.minecraft.launchwrapper.Launch;
+
+public class ModClassWriter extends ClassWriter {
+	public ModClassWriter(int flags) {
+		super(flags);
+	}
+
+	public ModClassWriter(ClassReader classReader, int flags) {
+		super(classReader, flags);
+	}
+
+	protected String getCommonSuperClass(final String type1, final String type2) {
+		Class<?> c, d;
+		ClassLoader classLoader = Launch.classLoader;

```

Comment:
In our use case this override is unnecessary. ClassWriter should be loaded by Launch.classLoader already.